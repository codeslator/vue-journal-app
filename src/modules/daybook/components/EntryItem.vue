<template>
  <div
    class="card mb-2 border-success entry-card pointer"
    @click="$router.push({ name: 'entry', params: { id: 1 } })"
  >
    <div class="card-body">
      <h5 class="card-title text-success">Hola</h5>
      <p class="card-subtitle mb-2 text-muted">{{ new Date() }}</p>
      <div class="card-text">
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Doloremque
        dolor sed doloribus sequi expedita! Vel veritatis deleniti quaerat eum
        doloribus.
      </div>
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.entry-card {
  &:hover {
    background-color: lighten($color: grey, $amount: 40);
    transition: 0.2s all ease-in;
  }
}
</style>