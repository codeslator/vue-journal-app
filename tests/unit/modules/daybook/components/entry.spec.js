import { shallowMount } from '@vue/test-utils';
import EntryItem from '@/modules/daybook/components/EntryItem';
import { daybookState } from '../../../mocks/daybook-state';

describe('Entry Component', () => {
  let wrapper;
  const mockRouter = {
    push: jest.fn(),
  }; // Mock del router

  beforeEach(() => {
    wrapper = shallowMount(EntryItem, {
      props: {
        entry: daybookState.entries[0],
      },
      global: { // Configuracion global
        mocks: {
          $router: mockRouter,
        }
      }
    })
  });


  test('Should match with last snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Should to redirect to click in entry-card', () => {
    const entryCard = wrapper.find('.entry-card');
    entryCard.trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: daybookState.entries[0].id
      }
    })
  });

  test('Should test computed properties', () => {
    expect(wrapper.vm.day).toBe(3);
    expect(wrapper.vm.month).toBe('Noviembre');
    expect(wrapper.vm.yearDay).toBe('2021, Mi√©rcoles');
  });
})