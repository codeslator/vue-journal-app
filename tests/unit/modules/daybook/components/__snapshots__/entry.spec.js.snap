// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Component Should match with last snapshot 1`] = `
<div class="card mb-2 border-success entry-card pointer">
  <div class="card-body">
    <h5 class="card-title text-success">Noviembre 3</h5>
    <p class="card-subtitle mb-2 text-muted">2021, Mi√©rcoles</p>
    <div class="card-text">Hola mundo</div>
  </div>
</div>
`;
